packages/flutter_tools/lib/src/fuchsia/fuchsia_device.dart

```diff
@@ -278,6 +295,95 @@ class FuchsiaDevice extends Device {
     throwToolExit('No ports found running $isolateName');
     return null;
   }
+
+  FuchsiaIsolateDiscoveryProtocol  getIsolateDiscoveryProtocol(String isolateName) => FuchsiaIsolateDiscoveryProtocol(this, isolateName);
+}
+
+class FuchsiaIsolateDiscoveryProtocol {
+  FuchsiaIsolateDiscoveryProtocol(this._device, this._isolateName, [
+    this._vmServiceConnector = _kDefaultFuchsiaIsolateDiscoveryConnector,
+    this._maxIterations = -1,
```

Comment:
Maybe leave a TODO to wire this up to something? Or remove it...

Reply:
This is a necessary evil for testing, but I don't think we want wire it up to the tool otherwise it becomes another timeout essentially.

Reply:
I could instead change this to a run once flag instead of max iter.